docs/core_docs/docs/integrations/chat/litellm_proxy.mdx

```diff
@@ -0,0 +1,157 @@
+---
+sidebar_label: LiteLLM Proxy
+---
+
+import CodeBlock from "@theme/CodeBlock";
+
+# LiteLLM Proxy Server
+
+Use [LiteLLM Proxy](https://docs.litellm.ai/docs/simple_proxy) for:
+
+- Calling 100+ LLMs OpenAI, Azure, Vertex, Bedrock/etc using OpenAI format
+- Use Virtual Keys to Set Budgets & Track Usage
+
+This allows you to call 100+ LLMs using `ChatOpenAI`
+
+```js
+import { ChatOpenAI } from "@langchain/openai";
+
+const model = new ChatOpenAI({
+  modelName: "claude-3-5-sonnet-20240620", // swap this for "gpt-4o-mini", "gemini-1.5-pro", "claude-3-5-sonnet-20240620"
+  openAIApiKey: "sk-1234",
+}, {
+  basePath: "http://0.0.0.0:4000", // set basePath to LiteLLM Proxy server
+});
+
+const message = await model.invoke("Hi there!");
+
+console.log(message);
+```
```

Comment:
This should either be an imported codeblock from the `examples` directory, or convert this to a notebook (use `tslab` to run typescript in notebooks) and have the code in cells. This way we can verify the code can be executed & run a linter/formatter across it.

Same for the rest of the typescript code blocks below